mt-kahypar/partition/refinement/deterministic/deterministic_label_propagation.cpp

```diff
@@ -38,17 +37,19 @@
 
 namespace mt_kahypar {
 
-  template<typename TypeTraits>
-  bool DeterministicLabelPropagationRefiner<TypeTraits>::refineImpl(
+
+  template<typename TypeTraits, typename GainTypes>
+  bool DeterministicLabelPropagationRefiner<TypeTraits, GainTypes>::refineImpl(
           mt_kahypar_partitioned_hypergraph_t& hypergraph,
           const vec<HypernodeID>&,
           Metrics& best_metrics,
           const double) {
     PartitionedHypergraph& phg = utils::cast<PartitionedHypergraph>(hypergraph);
     Gain overall_improvement = 0;
     constexpr size_t num_buckets = utils::ParallelPermutation<HypernodeID>::num_buckets;
-    const size_t num_sub_rounds = context.refinement.deterministic_refinement.num_sub_rounds_sync_lp;
-    const size_t num_buckets_per_sub_round = parallel::chunking::idiv_ceil(num_buckets, num_sub_rounds);
+    size_t num_sub_rounds = context.refinement.deterministic_refinement.num_sub_rounds_sync_lp;
+
+    GainComputation gain_computation(context, true);
```

Comment:
The other refinement algorithms use a member variable for `GainComputation`. I think this should preferably be uniform.

Since the steiner tree gain computation actually contains allocations that can be reused, I think making it a member makes sense and should also be done for the deterministic LP